pipeline/build/build.go

```diff
@@ -30,16 +33,44 @@ func (Pipe) Description() string {
 func (Pipe) Run(ctx *context.Context) error {
 	for _, build := range ctx.Config.Builds {
 		log.WithField("build", build).Debug("building")
+		if err := checkMain(ctx, build); err != nil {
+			return err
+		}
 		if err := runPipeOnBuild(ctx, build); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func checkMain(ctx *context.Context, build config.Build) error {
+	var glob = build.Main
+	if !strings.HasSuffix(glob, "main.go") {
```

Comment:
You can also have a `main` function if you build a file that is not called `main.go`. Better check if the suffix is `.go` or something like that, otherwise look at it as a directory